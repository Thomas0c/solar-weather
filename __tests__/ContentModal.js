import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ContentModal from '../lib/js/app/components/contentModal';

import { Text }Â from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(
    <ContentModal
      visible
      content={<Text>Test component</Text>}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
