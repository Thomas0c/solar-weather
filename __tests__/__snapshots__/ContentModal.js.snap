// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onOrientationChange={undefined}
  onRequestClose={undefined}
  onShow={undefined}
  supportedOrientations={undefined}
  transparent={true}
  visible={true}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.7)",
          "height": "100.00%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100.00%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityLiveRegion={undefined}
      accessibilityTraits={undefined}
      accessibilityViewIsModal={undefined}
      accessible={undefined}
      collapsable={undefined}
      hitSlop={undefined}
      importantForAccessibility={undefined}
      needsOffscreenAlphaCompositing={undefined}
      onAccessibilityTap={undefined}
      onLayout={undefined}
      onMagicTap={undefined}
      pointerEvents={undefined}
      removeClippedSubviews={undefined}
      renderToHardwareTextureAndroid={undefined}
      shouldRasterizeIOS={undefined}
      style={undefined}
      testID={undefined}
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityLiveRegion={undefined}
    accessibilityTraits={undefined}
    accessibilityViewIsModal={undefined}
    accessible={undefined}
    collapsable={undefined}
    hitSlop={undefined}
    importantForAccessibility={undefined}
    needsOffscreenAlphaCompositing={undefined}
    onAccessibilityTap={undefined}
    onLayout={undefined}
    onMagicTap={undefined}
    pointerEvents={undefined}
    removeClippedSubviews={undefined}
    renderToHardwareTextureAndroid={undefined}
    shouldRasterizeIOS={undefined}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#F0F0F0",
        "height": "55.00%",
        "marginTop": "15.00%",
        "position": "relative",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 5,
        "width": "85.00%",
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Test component
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignSelf": "center",
            "bottom": -20,
            "position": "absolute",
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityLiveRegion={undefined}
        accessibilityTraits={undefined}
        accessibilityViewIsModal={undefined}
        accessible={undefined}
        collapsable={undefined}
        hitSlop={undefined}
        importantForAccessibility={undefined}
        needsOffscreenAlphaCompositing={undefined}
        onAccessibilityTap={undefined}
        onLayout={undefined}
        onMagicTap={undefined}
        pointerEvents={undefined}
        removeClippedSubviews={undefined}
        renderToHardwareTextureAndroid={undefined}
        shouldRasterizeIOS={undefined}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E77F6D",
            "borderRadius": 20,
            "borderWidth": 0,
            "height": 40,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 40,
          }
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          adjustsFontSizeToFit={undefined}
          allowFontScaling={true}
          ellipsizeMode="tail"
          minimumFontScale={undefined}
          numberOfLines={undefined}
          onLayout={undefined}
          onLongPress={undefined}
          onPress={undefined}
          pressRetentionOffset={undefined}
          selectable={undefined}
          selectionColor={undefined}
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "avenir",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
          suppressHighlighting={undefined}
          testID={undefined}
          textBreakStrategy={undefined}
        >
          X
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
